src/main/scala/dpla/ingestion3/model/DplaMapData.scala

```diff
@@ -0,0 +1,145 @@
+  * Contains type definitions that express cardinality of fields
+  * and a union type for fields that can be literals or uris.
+  */
+
+object DplaMapData {
+  type ZeroToMany[T] = Seq[T]
+  type AtLeastOne[T] = Seq[T]
+  type ZeroToOne[T] = Option[T]
+  type ExactlyOne[T] = T
+  type LiteralOrUri = Either[String,URI]
+}
+
+/**
+  * Container for the classes that represent an item in DPLA MAP.
+  *
+  * @see https://dp.la/info/developers/map/
+  *
+  * @param sourceResource Metadata about the source item rather than it's representations.
+  * @param edmWebResource Metadata about the item's representation on the provider's site.
+  * @param oreAggregation Metadata about the aggretation of the item across it's source and representations.
+  */
+case class DplaMapData(
+                         sourceResource: DplaSourceResource,
+                         edmWebResource: EdmWebResource,
+                         oreAggregation: OreAggregation
+                       )
+
+//Core Classes
+
+/**
+  * dpla:SourceResource is a subclass of "edm:ProvidedCHO," which comprises the described resources
+  * (in EDM called "cultural heritage objects") about which DPLA collects descriptions.
+  * It is here that attributes of the described resources are located,
+  * not the digital representations of them.
+  */
+case class DplaSourceResource(
+                               alternateTitle: ZeroToMany[String] = Seq(),
+                               collection: ZeroToMany[DcmiTypeCollection] = Seq(),
+                               contributor: ZeroToMany[EdmAgent] = Seq(),
+                               creator: ZeroToMany[EdmAgent] = Seq(),
+                               date: ZeroToMany[EdmTimeSpan] = Seq(),
+                               description: ZeroToMany[String] = Seq(),
+                               extent: ZeroToMany[String] = Seq(),
+                               format: ZeroToMany[String] = Seq(),
+                               genre: ZeroToMany[SkosConcept] = Seq(),
+                               identifier: ZeroToMany[String] = Seq(),
+                               language: ZeroToMany[SkosConcept] = Seq(),
+                               place: ZeroToMany[EdmPlace] = Seq(), //specified as dpla:Place in the spec
+                               publisher: ZeroToMany[EdmAgent] = Seq(),
+                               relation: ZeroToMany[LiteralOrUri] = Seq(),
+                               replacedBy: ZeroToMany[String] = Seq(),
+                               replaces: ZeroToMany[String] = Seq(),
+                               rights: AtLeastOne[String] = Seq(),
+                               rightsHolder: ZeroToMany[EdmAgent] = Seq(),
+                               subject: ZeroToMany[SkosConcept] = Seq(),
+                               temporal: ZeroToMany[EdmTimeSpan] = Seq(),
+                               title: AtLeastOne[String] = Seq(),
+                               `type`: ZeroToMany[String] = Seq() //should be URIs?
```

Comment:
It's probably OK to call this dcmiType or dctermsType to avoid having to quote it.